lib/logpipe.c

```diff
@@ -52,13 +52,19 @@ log_pipe_queue_slow_path(LogPipe *self, LogMessage *msg, const LogPathOptions *p
   if ((self->flags & PIF_SYNC_FILTERX_TO_MSG))
     filterx_eval_sync_message(path_options->filterx_context, &msg, path_options);
 
-  if (G_UNLIKELY(self->flags & (PIF_HARD_FLOW_CONTROL | PIF_JUNCTION_END | PIF_CONDITIONAL_MIDPOINT)))
+  if (G_UNLIKELY(self->flags &
+                 (PIF_HARD_FLOW_CONTROL | PIF_NO_HARD_FLOW_CONTROL | PIF_JUNCTION_END | PIF_CONDITIONAL_MIDPOINT)))
     {
       path_options = log_path_options_chain(&local_path_options, path_options);
       if (self->flags & PIF_HARD_FLOW_CONTROL)
         {
-          local_path_options.flow_control_requested = 1;
-          msg_trace("Requesting flow control", log_pipe_location_tag(self));
+          local_path_options.flow_control_requested = TRUE;
+          msg_trace("Enabling flow control", log_pipe_location_tag(self));
```

Comment:
another nitpick: we might want to add the message identifiers to this trace message to see where it was applied. again completely optional.